apps/bluetooth/app.js

```diff
@@ -55,14 +60,30 @@ module.exports = function (activity) {
 
   function speak (text, alternativeVoice) {
     logger.debug(`speak: ${text}`)
-    return activity.setForeground().then(() => {
-      if (wifi.getWifiState() === wifi.WIFI_CONNECTED) {
-        return activity.tts.speak(text, { impatient: false })
-      } else if (alternativeVoice != null) {
-        logger.debug('No wifi connection, play alternative voice.')
-        return activity.playSound(alternativeVoice)
-      }
-    }).then(() => afterSpeak())
+    if (!textIsEmpty(text)) {
+      return activity.setForeground().then(() => {
+        if (wifi.getWifiState() === wifi.WIFI_CONNECTED) {
+          return activity.tts.speak(text, { impatient: false })
+        } else if (alternativeVoice != null) {
+          logger.debug('No wifi connection, play alternative voice.')
+          return activity.playSound(alternativeVoice)
+        }
+      }).then(() => afterSpeak())
```

Comment:
1. No need to wrap a function with `afterSpeak()`
2. Need handle the exception?

Reply:
1. fixed.
2. catch exception.